clients/coredata/event.go

```diff
@@ -131,7 +131,20 @@ func (e *EventRestClient) EventsForDeviceAndValueDescriptor(deviceId string, vd
 
 // Add event
 func (e *EventRestClient) Add(event *models.Event, ctx context.Context) (string, error) {
-	return clients.PostRequest(e.url, []byte(event.String()), ctx)
+	content := clients.FromContext(clients.ContentType, ctx)
+
+	var data []byte
+	var err error
+	if content == clients.ContentTypeCBOR {
+		data, err = event.MarshalCBOR()
```

Comment:
See comment below on MarshalJSON. See also question on whether MarshalCBOR() is an interface method called by the library when marshaling or encoding. If it is not an interface method, then wrap the encoding in a "private" function within clients/coredata/event.go itself.